clis/ado-cli/src/commands/apim/apim-utils.ts

```diff
@@ -110,6 +110,32 @@ const _publishApiToApim = async ({
       finalParameters,
     );
 
+    const productsAlreadyInApi = await client.apiProduct.listByApis(
+      resourceGroupName,
+      apiManagementName,
+      name,
+    );
+    const productsToDelete: string[] = [];
+    for await (const p of productsAlreadyInApi) {
+      if (p.name && !products.includes(p.name)) {
+        productsToDelete.push(p.name);
+      }
+    }
+    await Promise.all(
+      productsToDelete.map(p =>
+        client.productApi
+          .delete(resourceGroupName, apiManagementName, name, p)
```

Comment:
Last two parameters to the function call are swapped. It's expecting `productId`, `apiId`. So it should be:
`(... p, name)` instead of `(... name, p) `

Reply:
Good catch! Fixed!